pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarConnectionFactory.java

```diff
@@ -1546,6 +1546,7 @@ private Reader<?> createReaderForBrowserForNonPartitionedTopic(
               .readerName("jms-queue-browser-" + UUID.randomUUID())
               .startMessageId(seekMessageId)
               .startMessageIdInclusive()
+              .subscriptionName(queueSubscriptionName)
```

Comment:
this is probably not going to work
because:
- the name of the subscription is unique, and you are going to use an existing name, the behavior is unpredictable (you have a Durable and a NonDurable subscription with the same name)
- you cannot run to QueueBrowsers concurrenly

II suggest to make this name configurable at PulsarConnectionFactory level, this is basically the name of the tool that you are using to monitor the topic and not the application itself.

Or alternatively we have to add a uuid suffix to this name:
- the name is still bound to the original queueSubcriptionName
- the name is unique